### PR TITLE
[MODULAR] Oppressive Force Relocation Medicell

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Adds in the Oppressive Force Relocation Medicell, when hit with it, security personnel will be instantly safely teleported back to security. This only works within the area of Medbay to avoid any abuse.

This Medicell comes standard with all Mediguns by default so that all medical doctors have an option to remove intrusive officers from their medbay.

## How This Contributes To The Skyrat Roleplay Experience
By giving doctors an option to remove possible intrusive factors that can interrupt or end RP, medical doctors can now have more room to properly treat and RP with their patients. This also makes security seriously consider whether or not they need to be present within medbay at any given time, discouraging them from over encroaching.

## Changelog

:cl:
add: Introduces the Oppressive Force Relocation Medicell, used to safely remove obstructive officers from medbay.
/:cl:

### CODE REVIEW

**Comment by softcerv** [comment]
I was originally going to have it also dispense a lolipop into the officers hand, but I don't want to encourage bad officers to run into medbay just for free lolipops

**Comment by Jezzared** [comment]
Personally, i like the lollipops. Very passive-agressive.
This is hilarious, but can we also get one for assistants that slaps them back to the medical lobby? I find metagangs more obtrusive than security in my own experience.

File: modular_skyrat/modules/modular_weapons/code/medcells.dm [hunk 1]
```diff
@@ -508,6 +508,47 @@
...
+
+	teleport_effect(officer.loc)
+
+	var/list/turf_list = list()
+
+	for(var/turf/turf_in_area in get_area_turfs(/area/security/brig))
+		if(!turf_in_area.is_blocked_turf())
+			turf_list += turf_in_area
+
+	officer.visible_message(span_notice("[officer] teleports to back to security, reestablishing a calm medbay enviorment!"))
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_weapons/code/medcells.dm]:
```suggestion
	officer.visible_message(span_notice("[officer] teleports to back to security, reestablishing a calm medbay environment!"))
```

**Review by Zonespace27 — COMMENTED** [review]
smh, cant evn grammer corect, smh

**Comment by softcerv** [comment]
> smh, cant evn grammer corect, smh

mean >:(

**Comment by softcerv** [comment]
> Personally, i like the lollipops. Very passive-agressive. This is hilarious, but can we also get one for assistants that slaps them back to the medical lobby? I find metagangs more obtrusive than security in my own experience.

what if we have it so they are sent to the brig with the security officers instead?

File: modular_skyrat/modules/modular_weapons/code/medcells.dm [hunk 2]
```diff
@@ -508,6 +508,47 @@
...
+	if(!(ACCESS_BRIG in target_access))
+		return FALSE
+
+	teleport_effect(officer.loc)
+
+	var/list/turf_list = list()
+
+	for(var/turf/turf_in_area in get_area_turfs(/area/security/brig))
+		if(!turf_in_area.is_blocked_turf())
+			turf_list += turf_in_area
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/medcells.dm]: I feel like we could afford to make that static instead, for better performance, ngl
```suggestion
	var/static/list/turf_list

	if(!turf_list)
		turf_list = list()
		for(var/turf/turf_in_area in get_area_turfs(/area/security/brig))
			if(!turf_in_area.is_blocked_turf())
				turf_list += turf_in_area
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Comment by Zonespace27** [comment]
Once april fools passes, make this more modifiable (have the area and required access be variables on the medicell) and I'd like to see this merged for adminbus purposes

**Comment by softcerv** [comment]
> Once april fools passes, make this more modifiable (have the area and required access be variables on the medicell) and I'd like to see this merged for adminbus purposes

oh yeah, I was going to make this into like an actual thing, and I can have this version be an admin one.

**Comment by softcerv** [comment]
With enough tweaking and work, this could be an actually incredibly useful tool for medical players. I'm going to draft this PR until I can move this into a more balanced state for regular gameplay.

**Comment by softcerv** [comment]
going to close this PR so that I can make a new one once this is in a finished state

**Pull request closed without merging**